generators/utils.js

```diff
@@ -133,60 +133,103 @@ function addEntityToMenu(generator, entityName, translationKey, className) {
 }
 
 function addEntityToRouterImport(generator, entityAngularName, fileName, folderName) {
-    jhipsterUtils.rewriteFile(
-        {
-            file: `${CLIENT_MAIN_SRC_DIR}/app/router/index.ts`,
-            needle: 'jhipster-needle-add-entity-to-router-import',
-            splicable: [generator.stripMargin(
-                // prettier-ignore
-                `|// prettier-ignore
-                |const ${entityAngularName} = () => import('../entities/${folderName}/${fileName}.vue');
-                |// prettier-ignore
-                |const ${entityAngularName}Update = () => import('../entities/${folderName}/${fileName}-update.vue');
-                |// prettier-ignore
-                |const ${entityAngularName}Details = () => import('../entities/${folderName}/${fileName}-details.vue');`
-            )]
-        },
-        generator
-    );
+    if (!generator.readOnly) {
+        jhipsterUtils.rewriteFile(
+            {
+                file: `${CLIENT_MAIN_SRC_DIR}/app/router/index.ts`,
+                needle: 'jhipster-needle-add-entity-to-router-import',
+                splicable: [generator.stripMargin(
+                    // prettier-ignore
+                    `|// prettier-ignore
+                    |const ${entityAngularName} = () => import('../entities/${folderName}/${fileName}.vue');
+                    |// prettier-ignore
+                    |const ${entityAngularName}Update = () => import('../entities/${folderName}/${fileName}-update.vue');
+                    |// prettier-ignore
+                    |const ${entityAngularName}Details = () => import('../entities/${folderName}/${fileName}-details.vue');`
+                )]
+            },
+            generator
+        );
+    } else {
+        jhipsterUtils.rewriteFile(
+            {
+                file: `${CLIENT_MAIN_SRC_DIR}/app/router/index.ts`,
+                needle: 'jhipster-needle-add-entity-to-router-import',
+                splicable: [generator.stripMargin(
+                    // prettier-ignore
+                    `|// prettier-ignore
+                    |const ${entityAngularName} = () => import('../entities/${folderName}/${fileName}.vue');
+                    |// prettier-ignore
+                    |const ${entityAngularName}Details = () => import('../entities/${folderName}/${fileName}-details.vue');`
+                )]
+            },
+            generator
+        );
```

Comment:
Here you can avoid some duplication too, at least by conditioning the content of the `splicable` value in a variable, and thus having something like this:
```
let splicable = `|// prettier-ignore
                    |const ${entityAngularName} = () => import('../entities/${folderName}/${fileName}.vue');`
if (!readOnly) {
     splicable += `|// prettier-ignore
                    |const ${entityAngularName}Update = () => import('../entities/${folderName}/${fileName}-update.vue');`
}
splicable += `|// prettier-ignore
                    |const ${entityAngularName}Details = () => import('../entities/${folderName}/${fileName}-details.vue');`

        jhipsterUtils.rewriteFile(
            {
                file: `${CLIENT_MAIN_SRC_DIR}/app/router/index.ts`,
                needle: 'jhipster-needle-add-entity-to-router-import',
                splicable: [generator.stripMargin(splicable)]
            },
            generator
        );
```